### ISSUE TITLE
Discrepancy of performance loss implementation

### ISSUE THREAD

**Issue by milliema** [xiangly55/LFME#1]
Thanks for the work!
在研究代码的过程中发现关于CE loss的计算，似乎原文公式与代码有些差别：
1）原文中对样本求取CEloss之后，再根据每个样本的vi进行加权求和；
![image](https://user-images.githubusercontent.com/22413367/119225845-dae5fb80-bb38-11eb-8332-7443c9e08f7e.png)
2）代码中直接用logits乘以vi，再求取CEloss。
![image](https://user-images.githubusercontent.com/22413367/119225857-e9ccae00-bb38-11eb-941a-ca0898590a6c.png)
二者在数学上并不等价，请问是否我理解错误？该如何解释？

**Comment by xiangly55** [comment]
Thanks for pointing out. Originally we use the former, and we empirically find that the latter yields slightly superior tail class performance. But overall, their performances are close.
